docker-entrypoint.sh

```diff
@@ -188,7 +188,7 @@ docker_process_sql() {
 # create initial database
 # uses environment variables for input: POSTGRES_DB
 docker_setup_db() {
-	if [ "$POSTGRES_DB" != 'postgres' ]; then
+	if [ "$POSTGRES_DB" != 'postgres' ] || [ "$POSTGRES_DB" != 'template1' ]; then
```

Comment:
The entrypoint already notes that `initdb` will create the three databases:

https://github.com/docker-library/postgres/blob/38841304ddd06dd3b39cc080b2c8fce5a5be8e52/docker-entrypoint.sh#L64

And the intent of this `"$POSTGRES_DB" != 'postgres' ` is to not create an already created database (since postgresql doesn't have an `IF NOT EXISTS` on create database), it might make more sense to key off the database name already existing?  (just in case `initdb` changes in the future)

Reply:
:grimacing: great point (and https://stackoverflow.com/a/18389184/433558 is depressing :disappointed:)

So maybe something like this?

```suggestion
	local exists
	if exists="$(POSTGRES_DB= docker_process_sql --dbname postgres --set db="$POSTGRES_DB" --tuples-only --command 'SELECT 1 FROM pg_database WHERE datname = :"db"')" && [ -z "$exists" ]; then
```

(Sorry it's not super obvious, but this will also need to be copied to every directory for the GitHub Actions to test it properly.)